### ISSUE TITLE
Module links not working on hackage

### ISSUE THREAD

**Issue by barischrooneyj** [well-typed/optics#237]
![Screenshot 2019-09-03 at 16 36 14](https://user-images.githubusercontent.com/6631452/64185849-1d037100-ce6e-11e9-8875-c0745cdbb0c6.png)

**Comment by adamgundry** [comment]
Yes, this is a consequence of Hackage not supporting `reexported-modules` very well unfortunately (see haskell/hackage-server#577). You can see the links in `optics-core` and `optics-extra`, but the reexports in `optics` don't get linked. I'm not sure if we can do much to improve matters...

**Comment by adamgundry** [comment]
We have a workaround for this in #465, but the script needs to be run and the results uploaded for each release, and it seems that wasn't done for the most recent release.

A worry is that `@since` pragmas in re-exported modules will be copied verbatim by the script, even though they apply to (e.g.) `optics-core` rather than `optics`.

**Comment by phadej** [comment]
> optics-core rather than optics.

That's a small worry, as `optics(-core)` versions are synchronized. If something is *since `optics-core-0.5.6` it won't be in `optics-0.4`, but only in `optics-0.5.6`. In particular, there will be `optics-0.5.6` to correspond to `optics-core-0.5.6` if the addition is in the re-exported module (and in particular visible in haddocks).
